src/main/java/apoc/convert/Convert.java

```diff
@@ -91,48 +91,69 @@ else if (list.getClass().isArray() && !list.getClass().getComponentType().isPrim
         return list == null ? null : new SetBackedList(new LinkedHashSet<>(list));
     }
     
-    @SuppressWarnings("rawtypes")
 	@UserFunction
-    @Description("apoc.convert.toListOf(value, type) | tries it's best to convert "
-    		+ "the value to a list of supported type, supported types are: "
-    		+ "int, long, double, String, boolean, node, relationship")
-    public List toListOf(@Name("list") Object list, @Name("type") String type) {
-        List objectList = convertToList(list);
+    @Description("apoc.convert.toIntList(value) | tries it's best to convert "
+    		+ "the value to a list of integers")
+    public List<Integer> toIntList(@Name("list") Object list) {
+        List<Object> objectList = convertToList(list);
```

Comment:
perhaps we can simplify all these functions by having a convertToList(list, Long.class)

that internally does the "list" to null or a stream and then a switch on the type for the element migration
also if the element is already of a type don't convert it again.

if it doesn't exist we should also add the functions toFloat(value) -> Double and toInteger(value) -> Long

```
Double toFloat(Object value) {
if (value==null) return null;
if (value instanceof Number) return ((Number)value).doubleValue();
try {
  return Double.parseDouble(value.toString);
} catch (NFE ) {
   return null;
}
}

```